src/axom/primal/geometry/BezierCurve.hpp

```diff
@@ -158,14 +228,45 @@ class BezierCurve
   /// Returns the order of the Bezier Curve
   int getOrder() const { return static_cast<int>(m_controlPoints.size()) - 1; }
 
-  /// Clears the list of control points
+  /// Make trivially rational. If already rational, do nothing
+  void makeRational()
+  {
+    if(!isRational())
+    {
+      const int ord = getOrder();
+      m_weights.resize(ord + 1);
+      for(int i = 0; i < ord + 1; i++) m_weights[i] = 1.0;
+    }
+  }
+
+  /// Make nonrational by shrinking array of weights
+  void makeNonrational() { m_weights.resize(0); }
+
+  /// Use array size as flag for rationality
+  bool isRational() const { return (m_weights.size() != 0); }
+
+  /// Check that the weights used are valid
+  bool isValid() const
+  {
+    if(isRational()) return true;
```

Comment:
It looks like this method checks that a *rational* Bezier curve is valid.  Perhaps rename the method to `isValidRational` or similar.

Further, the test at line 251 seems to be inverted. Non-rational curves would then go on to pull values out of `m_weights`.  In addition to inverting the test at line 251, consider adding a check before the loop at line 254 that `m_weights` is the same length as `m_controlPoints`.

Reply:
Thanks for catching that, that check is definitely backwards as it is now.